### DISCUSSION TITLE
Aggregate Point Observations

### DISCUSSION THREAD

**Discussion by cjmelick** [dtcenter/METplus#1767] [File I/O] [answered]
I am trying to aggregate point observations from multiple netCDF files created from ascii2nc into one before running point2grid.   Is there a method/tool option within MET to accomplish this task or do I need to do this on my own? 

Thanks,
Chris Melick

**Comment by j-opatz** [comment]
Hi Chris,

And thank you for your question. Unfortunately, there seems to be no straightforward answer, at least not using the MET tools. Currently, point2grid only accepts 1 input file; a discussion with John HG has us in agreement that this is an opportunity to improve the point2grid tool to mimic point_stat's behavior, which can ingest multiple input files from the `-point_obs` option. 

I hope to write up a Github issue in the very near future to capture this need and link it in this Discussions.

For the time being, there's a workaround you could try:

Currently, the [read_met_point_obs.py](https://github.com/dtcenter/MET/blob/develop/scripts/python/read_met_point_obs.py) script reads in point observations; so you might be able to run ascii2nc with Python Embedding to ingest multiple input files and output 1 netCDF file. John HG and I aren't sure if this would work for certain, but it could be a (temporary) solution!

Another option is using the `ncks` function outside of MET and METplus. Running that function with a simple shell script would easily combine all of your ascii2nc output into 1 netCDF. However, as mentioned, this is a separate function outside of MET and doesn't fit your question's parameters. But it would definitely get the job done.

**Reply by cjmelick** [reply]
What are the maximum number of observations that can be in an ascii file for the ascii2nc program to safely work and not provide some sort of segmentation fault error?

**Reply by cjmelick** [reply]
And by extension, I'm curious about the same question when using point2grid.

**Reply by j-opatz** [reply]
That's a good question! I'll ping @JohnHalleyGotway for the specific answer, since the code will probably determine that limit.

**Comment by cjmelick** [comment]
Thank you very much for the suggested alternatives as temporary solutions.  I will manage just fine with some workarounds until the functionality exists inside of MET is possible.   If John HG or you could please let me know when that comes about, I would appreciate it!

Chris

**Comment by j-opatz** [comment, marked as answer]
Hi Chris,

I've just created an issue for this here: https://github.com/dtcenter/MET/issues/2248 

John HG and I can have a further discussion on where this best fits into the development timeline, so for now it'll reside in the backlog of development ideas to conserve on current funding and time.

**Reply by cjmelick** [reply]
Thank you very much!  No worries, I have a workaround in place right now.

--Chris
